include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -628,24 +628,27 @@ struct __early_exit_find_or
             using _ShiftedIdxType =
                 typename ::std::conditional<_OrTagType::value, decltype(__init_index + __i * __shift),
                                             decltype(__found_local.load())>::type;
-            _IterSize __current_iter = oneapi::dpl::__internal::__invoke_if_else(
-                _BackwardTagType{}, [__n_iter, __i]() { return __n_iter - 1 - __i; }, [__i]() { return __i; });
+
+            _IterSize __current_iter =  [__n_iter, __i](){ 
+                if constexpr (_BackwardTagType::value)
+                    return __n_iter - 1 - __i;
+                else
+                    return __i;
+            }();
 
             _ShiftedIdxType __shifted_idx = __init_index + __current_iter * __shift;
             // TODO:[Performance] the issue with atomic load (in comparison with __shifted_idx for early exit)
             // should be investigated later, with other HW
             if (__shifted_idx < __n && __pred(__shifted_idx, __rngs...))
             {
-                oneapi::dpl::__internal::__invoke_if_else(
-                    _OrTagType{}, [&__found_local]() { __found_local.store(1); },
-                    [&__found_local, &__comp, &__shifted_idx]() {
-                        for (auto __old = __found_local.load(); __comp(__shifted_idx, __old);
-                             __old = __found_local.load())
-                        {
-                            __found_local.compare_exchange_strong(__old, __shifted_idx);
-                        }
-                    });
-                return;
+                if constexpr (_OrTagType::value)
+                    __found_local.store(1);
+                else
+                    for (auto __old = __found_local.load(); __comp(__shifted_idx, __old);
+                         __old = __found_local.load())
+                    {
+                        __found_local.compare_exchange_strong(__old, __shifted_idx);
+                    }
```

Comment:
I'd prefer `{ }` around multi-line statements even if they are single statements, whether it be loops or just very long lines broken by formatting.

Reply:
I agree. I just changed all of these to add more braces.

Reply:
Great, thanks!